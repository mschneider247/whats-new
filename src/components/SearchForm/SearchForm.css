header {
  display: flex;
  height: 120px;
  padding: 5% 0;
}

input {
  height: 40px;
  width: 200px;
  margin: 0 2% 0 40%;
}

button {
  height: 40px;
}

