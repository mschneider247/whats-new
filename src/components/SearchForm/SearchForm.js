import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {search: ''}
  }

  handleSearch = event => {
    this.setState({ [event.target.name]: event.target.value})
  }

  render() {
    return (
      <header>
        <h1>What's New?</h1>
        <input
          type="text"
          placeholder="Search"
          name="search"
          value={this.state.search}
          onChange={event => this.handleSearch(event)} 
        />
        <button>Search News</button>
      </header>
    )
  }
}

export default SearchForm;