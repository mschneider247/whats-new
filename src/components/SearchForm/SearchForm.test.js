import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SearchForm
      searchNews={jest.fn()}
      />);
      expect(wrapper).toMatchSnapshot();

    });
});