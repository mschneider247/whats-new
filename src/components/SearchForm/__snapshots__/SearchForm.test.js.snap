// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<header>
  <h1>
    What's New?
  </h1>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search News
  </button>
</header>
`;
